src/audio/codec_adapter/codec_adapter.c

```diff
@@ -192,19 +194,62 @@ int codec_adapter_prepare(struct comp_dev *dev)
 		return -EINVAL;
 	}
 
+	/* allocate memory for input buffers */
+	mod->input_buffers = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
```

Comment:
this appears to be duplicated between input vs output, maybe make a "allocat buffers" function?

Reply:
@cujomalainey the only thing I'd be able to save is maybe 4 lines the struct definitions for input_buffers and out_buffers are different, so they will have to remain as is. Do you still want be to create a new function for allocating input_buffers and output_buffers?

Reply:
If this is the only possible place this code can be used then this is fine, if there is a chance it can be used elsewhere we should extract.

Reply:
this will be the only place where it will be used @cujomalainey.